components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/inbound/InboundAuthConfigToApiModel.java

```diff
@@ -0,0 +1,105 @@
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.api.server.application.management.v1.core.functions.application.inbound;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.wso2.carbon.identity.api.server.application.management.common.ApplicationManagementConstants;
+import org.wso2.carbon.identity.api.server.application.management.v1.InboundProtocolListItem;
+import org.wso2.carbon.identity.api.server.application.management.v1.core.functions.Utils;
+import org.wso2.carbon.identity.api.server.common.ContextLoader;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.application.common.model.InboundAuthenticationConfig;
+import org.wso2.carbon.identity.application.common.model.InboundAuthenticationRequestConfig;
+import org.wso2.carbon.identity.application.common.model.ServiceProvider;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.wso2.carbon.identity.api.server.application.management.common.ApplicationManagementConstants.APPLICATION_MANAGEMENT_PATH_COMPONENT;
+import static org.wso2.carbon.identity.api.server.application.management.common.ApplicationManagementConstants.INBOUND_PROTOCOLS_PATH_COMPONENT;
+import static org.wso2.carbon.identity.api.server.common.Constants.V1_API_PATH_COMPONENT;
+
+/**
+ * Converts Inbound Authentication configuration of an application into the API model.
+ */
+public class InboundAuthConfigToApiModel implements Function<ServiceProvider, List<InboundProtocolListItem>> {
+
+    @Override
+    public List<InboundProtocolListItem> apply(ServiceProvider application) {
+
+        String applicationId = application.getApplicationResourceId();
+        InboundAuthenticationConfig inboundAuthConfig = application.getInboundAuthenticationConfig();
+
+        if (inboundAuthConfig != null) {
+            if (ArrayUtils.isNotEmpty(inboundAuthConfig.getInboundAuthenticationRequestConfigs())) {
+
+                List<InboundProtocolListItem> inboundProtocolListItems = new ArrayList<>();
+                Arrays.stream(inboundAuthConfig.getInboundAuthenticationRequestConfigs()).forEach(
+                        inbound -> inboundProtocolListItems.add(buildInboundProtocolListItem(applicationId, inbound)));
+
+                return inboundProtocolListItems;
+            }
+        }
+
+        return Collections.emptyList();
+    }
+
+    private InboundProtocolListItem buildInboundProtocolListItem(String applicationId,
+                                                                 InboundAuthenticationRequestConfig inbound) {
+
+        return new InboundProtocolListItem()
+                // TODO: add the friendly name from metadata service.
```

Comment:
shall we remove todos

Reply:
TODOs will be addressed in the next PR for application management REST APIs
Created https://github.com/wso2/product-is/issues/7000 to track all the TODOs